allensdk/brain_observatory/ecephys/stimulus_analysis/drifting_gratings.py

```diff
@@ -115,7 +140,30 @@ def number_contrast(self):
     def null_condition(self):
         """ Stimulus condition ID for null (blank) stimulus """
         return self.stimulus_conditions[self.stimulus_conditions[self._col_tf] == 'null'].index
-    
+
+    @property
+    def stimulus_conditions_contrast(self):
+        """ Stimulus conditions for contrast stimulus """
+        if self._stimulus_conditions_contrast is None:
+            contrast_condition_list = self.stim_table_contrast.stimulus_condition_id.unique()
```

Comment:
Ignore this if this method is quick:
I think we can set() this to run the filter on the next line in O(n) vs O(n**2).

Reply:
Technically it will be O(n*m) depending on the number of drifting_gratings_contrast conditions. But I don't want to say how efficient it is because I've not been able to profile it with real data that has "drifting_gratings_contrast" (if at some point you want to consider using asv with the AllenSDK I can help).